flash/data/data_module.py

```diff
@@ -0,0 +1,354 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+import platform
+from copy import deepcopy
+from typing import Any, Callable, Optional, Union
+
+import pytorch_lightning as pl
+import torch
+from pytorch_lightning.core.datamodule import _DataModuleWrapper, track_data_hook_calls
+from pytorch_lightning.trainer.states import RunningStage
+from pytorch_lightning.utilities import rank_zero_only
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from torch.utils.data import DataLoader, Dataset
+from torch.utils.data.dataset import Subset
+
+from flash.data.auto_dataset import AutoDataset
+from flash.data.data_pipeline import DataPipeline, Postprocess, Preprocess
+
+
+class MockLightningModule(pl.LightningModule):
+
+    pass
+
+
+class TaskDataPipeline(DataPipeline):
+
+    def per_batch_transform(self, batch: Any) -> Any:
+        return (batch["x"], batch.get('target', batch.get('y'))) if isinstance(batch, dict) else batch
+
+
+class _FlashDataModuleWrapper(_DataModuleWrapper):
```

Comment:
```suggestion
class _FlashDataModuleWrapper(type):
```
Don't see a need of creating a subclass of `_DataModuleWrapper`.

Reply:
The IDE complains.

<img width="659" alt="Screenshot 2021-03-22 at 18 37 38" src="https://user-images.githubusercontent.com/12861981/112040927-af955f00-8b3d-11eb-874c-37afd088aa56.png">

Reply:
It seems it is necessary as DataModule from Lightning as a metaclass

<img width="1284" alt="Screenshot 2021-03-22 at 18 38 28" src="https://user-images.githubusercontent.com/12861981/112041018-cc319700-8b3d-11eb-8ae9-c08af92ab362.png">